test/Autofac.Test/TypeExtensionsTests.cs

```diff
@@ -41,5 +42,176 @@ public void AnOpenGenericTypeIsNotAClosedTypeOfAnything()
         {
             Assert.False(typeof(CommandBase<>).IsClosedTypeOf(typeof(CommandBase<>)));
         }
+
```

Comment:
We didn't really have any test coverage specifically around these extensions so I added it to ensure the update to the constructor search flags didn't break anything else.